internal/controller/component/controller.go

```diff
@@ -72,28 +66,38 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{}, err
 	}
 
-	previousCondition := meta.FindStatusCondition(component.Status.Conditions, controller.TypeCreated)
+	// Keep a copy of the original object for comparison
```

Comment:
```suggestion
       // Reconciliation should be avoided if the component is already created.
       if shouldIgnoreReconcile(component) {
                r.Recorder.Event(component, corev1.EventTypeNormal, "ComponentCreated", "Successfully created "+component.Name)
		return ctrl.Result{}, nil
	}
	// Keep a copy of the original object for comparison
```

func shouldIgnoreReconcile(component *choreov1.Component) bool {
	return meta.FindStatusCondition(component.Status.Conditions, string(controller.TypeCreated)) != nil 
}

We can remove the below part

Reply:
This cannot be done since updates are coming.